packages/eslint-plugin/src/rules/unbound-method.ts

```diff
@@ -250,7 +255,13 @@ function isSafeUse(node: TSESTree.Node): boolean {
       return ['instanceof', '==', '!=', '===', '!=='].includes(parent.operator);
 
     case AST_NODE_TYPES.AssignmentExpression:
-      return parent.operator === '=' && node === parent.left;
+      return (
+        parent.operator === '=' &&
+        (node === parent.left ||
+          (allowSuper &&
+            (node as TSESTree.MemberExpression) &&
```

Comment:
This is unsafe - prefer an explicit check.

```suggestion
            node.type === AST_NODE_TYPES.MemberExpression &&
```